Alpha_shapes_3/include/CGAL/internal/Lazy_alpha_nt_3.h

```diff
@@ -73,27 +105,26 @@ struct Types_for_alpha_nt_3< ::CGAL::Tag_true,Input_traits,Kernel_input,Kernel_a
 //Converter types
   typedef CGAL::Cartesian_converter<Kernel_input,Kernel_approx>   To_approx;
   typedef CGAL::Cartesian_converter<Kernel_input,Kernel_exact>    To_exact;
-//Traits types
-  typedef ::CGAL::Regular_triangulation_euclidean_traits_3<Kernel_approx>                       Approx_traits;
-  typedef ::CGAL::Regular_triangulation_euclidean_traits_3<Kernel_exact>                        Exact_traits;
 //Point types
-  typedef typename Approx_traits::Weighted_point Approx_point;
-  typedef typename Exact_traits::Weighted_point  Exact_point;
+  typedef typename Kernel_approx::Weighted_point_3 Approx_point;
+  typedef typename Kernel_exact::Weighted_point_3  Exact_point;
   typedef typename Input_traits::Weighted_point_3  Input_point;
 //Constructions 
-  typedef typename Approx_traits::Compute_squared_radius_smallest_orthogonal_sphere_3           Approx_squared_radius;
-  typedef typename Exact_traits::Compute_squared_radius_smallest_orthogonal_sphere_3            Exact_squared_radius; 
+  typedef typename Kernel_approx::Compute_squared_radius_smallest_orthogonal_sphere_3           Approx_squared_radius;
+  typedef typename Kernel_exact::Compute_squared_radius_smallest_orthogonal_sphere_3            Exact_squared_radius;
 };
 
 
-template<class Input_traits, class Kernel_input, bool mode, class Weighted_tag>
+template<class Input_traits, bool mode, class Weighted_tag>
 class Lazy_alpha_nt_3{
 //NT & kernels
   typedef CGAL::Interval_nt<mode>                                                               NT_approx;
   //Gmpq or Quotient<MP_float>
   typedef Exact_field_selector<double>::Type                                                    NT_exact;
   typedef CGAL::Simple_cartesian<NT_approx>                                                     Kernel_approx;
   typedef CGAL::Simple_cartesian<NT_exact>                                                      Kernel_exact;
+  typedef typename Kernel_traits<typename Input_traits::Point_3>::Kernel   Kernel_input;
+
```

Comment:
No, because the traits class might not derive from a kernel, and thus it might not know what e.g. a `Circle_3` is. Then, `Cartesian_converter<Traits, EK>` will not compile since it won't know how to compile `EK::Circle_3 operator()(const Traits::Circle_3)`.